src/ciphers/aes_cbc.rs

```diff
@@ -1,141 +1,81 @@
-// Copyright 2020 IOTA Stiftung
+// Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
 use core::{
     marker::PhantomData,
     num::NonZeroUsize,
-    ops::{Shl, Sub},
+    ops::{Shl, Shr},
 };
 
-use aes_crate::{Aes128, Aes192, Aes256, BlockCipher, NewBlockCipher};
-use block_modes::{block_padding::Pkcs7, BlockMode, Cbc};
+use aes::{
+    cipher::{
+        block_padding::{Padding, Pkcs7},
+        BlockCipher, BlockDecryptMut, BlockEncryptMut, BlockSizeUser, InnerIvInit, KeyInit, KeySizeUser,
+    },
+    Aes128, Aes192, Aes256,
+};
+use cbc::{Decryptor as CbcDecryptor, Encryptor as CbcEncryptor};
 use generic_array::{
-    sequence::Split,
-    typenum::{Double, Unsigned, B1, U16, U24, U32},
+    typenum::{Unsigned, B1},
     ArrayLength,
 };
-use hmac_::{
-    digest::{
-        block_buffer::Eager,
-        core_api::{BlockSizeUser, BufferKindUser, CoreProxy, FixedOutputCore, UpdateCore},
-        typenum::{IsLess, Le, NonZero, U256},
-        FixedOutput, HashMarker, OutputSizeUser, Reset,
-    },
-    Hmac, Mac,
+use hmac::{
+    digest::{Digest, FixedOutput, OutputSizeUser, Update},
+    SimpleHmac,
 };
 use sha2::{Sha256, Sha384, Sha512};
 use subtle::ConstantTimeEq;
+use zeroize::Zeroize;
 
 use crate::ciphers::traits::{Aead, Key, Nonce, Tag};
 
 /// AES-CBC using 128-bit key and HMAC SHA-256.
-pub type Aes128CbcHmac256 = AesCbc<Aes128, Sha256, U16, U16>;
+pub type Aes128CbcHmac256 = CbcHmac<Aes128, Pkcs7, Sha256>;
```

Comment:
Out of curiosity, was `AesCbc` previously automatically using `Pkcs7`? Or why do we need to put it now?

Reply:
Yes, for some reason, although there could be other paddings. Also, it was named AesCbc but it was possible to use ciphers other than AES. Having padding as generic argument just feels natural in the current implementation.

Reply:
Thanks